QueryCacheMapWideEventsTest

**com.hazelcast.map.impl.querycache.QueryCacheMapWideEventsTest.no_event_lost_during_migrations__with_many_parallel_nodes**
```java
java.lang.AssertionError: expected:<0> but was:<3316>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at com.hazelcast.map.impl.querycache.QueryCacheMapWideEventsTest.no_event_lost_during_migrations(QueryCacheMapWideEventsTest.java:112)
	at com.hazelcast.map.impl.querycache.QueryCacheMapWideEventsTest.no_event_lost_during_migrations__with_many_parallel_nodes(QueryCacheMapWideEventsTest.java:69)
```
https://hazelcast-l337.ci.cloudbees.com/job/new-lab-fast-pr/14777/testReport/junit/com.hazelcast.map.impl.querycache/QueryCacheMapWideEventsTest/no_event_lost_during_migrations__with_many_parallel_nodes/

--- a/hazelcast/src/test/java/com/hazelcast/map/impl/querycache/QueryCacheMapWideEventsTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/map/impl/querycache/QueryCacheMapWideEventsTest.java
@@ -56,9 +56,7 @@ public class QueryCacheMapWideEventsTest extends HazelcastTestSupport {
     private static final String QUERY_CACHE_NAME = "cacheName";
     private static final String PARTITION_COUNT = "1999";
     private static final int TEST_DURATION_SECONDS = 3;
-    private static final int TEN_MINUTES_TIMEOUT = 60000 * 10;
 
-    private AtomicInteger eventLostCounter = new AtomicInteger();
     private TestHazelcastInstanceFactory factory = createHazelcastInstanceFactory();
 
     @Test
@@ -66,19 +64,24 @@ public class QueryCacheMapWideEventsTest extends HazelcastTestSupport {
         no_event_lost_during_migrations(1, 0);
     }
 
-    @Test(timeout = TEN_MINUTES_TIMEOUT)
+    @Test
     @Category(SlowTest.class)
     public void no_event_lost_during_migrations__with_many_parallel_nodes() {
         assertTrueEventually(new AssertTask() {
             @Override
             public void run() {
-                no_event_lost_during_migrations(3, 5);
+                try {
+                    no_event_lost_during_migrations(3, 5);
+                } finally {
+                    factory.shutdownAll();
+                }
             }
         });
     }
 
     private void no_event_lost_during_migrations(int parallelNodeCount, int startNewNodeAfterSeconds) {
-        newQueryCacheOnNewNode();
+        final AtomicInteger eventLostCounter = new AtomicInteger();
+        newQueryCacheOnNewNode(eventLostCounter);
 
         List<Thread> threads = new ArrayList<Thread>();
         for (int i = 0; i < parallelNodeCount; i++) {
@@ -86,7 +89,7 @@ public class QueryCacheMapWideEventsTest extends HazelcastTestSupport {
             Thread doMapClear = new Thread() {
                 @Override
                 public void run() {
-                    IMap map = newQueryCacheOnNewNode();
+                    IMap map = newQueryCacheOnNewNode(eventLostCounter);
 
                     long endMillis = System.currentTimeMillis() + SECONDS.toMillis(TEST_DURATION_SECONDS);
                     while (System.currentTimeMillis() < endMillis) {
@@ -143,17 +146,16 @@ public class QueryCacheMapWideEventsTest extends HazelcastTestSupport {
         }
     }
 
-    private IMap newQueryCacheOnNewNode() {
+    private IMap newQueryCacheOnNewNode(final AtomicInteger eventLostCounter) {
         HazelcastInstance node = factory.newHazelcastInstance(newConfig());
         waitClusterForSafeState(node);
         IMap map = node.getMap(MAP_NAME);
-        QueryCache queryCache = map.getQueryCache(QUERY_CACHE_NAME);
-        addEventLostListenerToQueryCache(queryCache);
+        addEventLostListenerToQueryCache(map, eventLostCounter);
         return map;
     }
 
-    private void addEventLostListenerToQueryCache(QueryCache queryCache) {
-        queryCache.addEntryListener(new EventLostListener() {
+    private void addEventLostListenerToQueryCache(IMap map, final AtomicInteger eventLostCounter) {
+        map.getQueryCache(QUERY_CACHE_NAME).addEntryListener(new EventLostListener() {
             @Override
             public void eventLost(EventLostEvent event) {
                 eventLostCounter.incrementAndGet();

